packages/mui-joy/src/styles/extendTheme.ts

```diff
@@ -606,6 +606,7 @@ export default function extendTheme(themeOptions?: CssVarsThemeOptions): Theme {
     getCssVar,
     spacing: getSpacingVal(spacing),
     font: { ...prepareTypographyVars(mergedScales.typography), ...mergedScales.font },
+    internal_cache: {},
```

Comment:
I would have expected to have to add `.internal_cache` to the system's `createTheme` function only, but I ended up needing to add it at various locations. It would be neat to pass all our themes through that function to avoid repeating internal details in various places.

Reply:
Why couldn't we just add it in the `createTheme` function? What was the issue?

Reply:
Some created theme objects don't go through system's `createTheme` at all. I don't feel confident enough to pass them all through it at this point, I'm not sure what edge-cases I might find doing so.